pkg/services/govmomi/service.go

```diff
@@ -37,11 +37,18 @@ import (
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/govmomi/extra"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/govmomi/net"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/govmomi/task"
 	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/util"
 )
 
+const (
+	maxTaskCount = 80
+)
+
 // VMService provdes API to interact with the VMs using govmomi
-type VMService struct{}
+type VMService struct {
+	TaskCounter *task.Counter
```

Comment:
This is definitely a good start for introducing rate-limiting.

However, I think we should document the scope of this task counter, which, if I got this right, is responsible for rate-limiting everything managed by this controller.

In the future might be we should be required to improve this by making CAPV aware of the overall task pressure of the underlying vSphere infrastructure (a vSphere infrastructure can have pressure from something external from CAPV), otherwise, this mechanism could not limit enough depending on the conditions.

Vice versa, CAPV pressure could be spread on many vSphere infrastructures so this mechanism could be limiting too much depending on the conditions.

Documenting the current scope could help us in managing future improvements if required.